yktracec/src/jitmodbuilder.cc

```diff
@@ -1912,6 +1922,54 @@ class JITModBuilder {
     finalise(AOTMod, &Builder);
     return JITMod;
   }
+
+  /// Handle promoted variables.
+  void handlePromote(CallInst *CI, MappableFrame *MPF, BasicBlock *BB,
+                     size_t CurBBIdx, size_t CurInstrIdx) {
+    // First lookup the constant value the trace is going to use.
+    uint64_t PConst = __yk_lookup_promote_usize();
+    Value *PConstLL =
+        Builder.getIntN(PointerSizedIntTy->getIntegerBitWidth(), PConst);
+    Value *PromoteVar = CI->getArgOperand(0);
+    Value *JITPromoteVar = getMappedValue(PromoteVar);
+
+    // Now emit a guard for the value, so that we deoptimise if the
+    // value ever changes.
```

Comment:
I think this should be "so that we deoptimise if a later execution of the trace sees a different value for this `promote`" or similar.

Reply:
I tried a rewording in 96f196064a3dafea592a5e5a3da4c750e1a06e0a. Any good?